apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-override-save-button.ts

```diff
@@ -0,0 +1,127 @@
+		};
+		const overrideSaveButtonClick = ( selector: string ) => {
+			const button = document.querySelector( selector );
+			if ( button ) {
+				button.textContent = __( 'Upgrade now', 'wpcom-live-preview' );
+				button.addEventListener( 'click', saveButtonClickHandler );
+			}
+		};
+
+		/**
+		 * This overrides the tooltip text for the save button.
+		 *
+		 * The tooltip is shown after a delay.
+		 * So we observe the DOM changes to avoid being fragile to the delay.
+		 * The observer is activated only when the user hovers over the button to optimize performance.
+		 */
+		const observer = new MutationObserver( ( mutations ) => {
+			mutations.forEach( ( mutation ) => {
+				mutation.addedNodes.forEach( ( node ) => {
+					if ( node.nodeType === Node.ELEMENT_NODE ) {
+						const tooltip = ( node as Element ).querySelector( '.components-tooltip' );
+						if ( tooltip ) {
+							tooltip.textContent = __( 'Upgrade now', 'wpcom-live-preview' );
+						}
+					}
+				} );
+			} );
+		} );
+		const startObserver = () => {
+			observer.observe( document.body, { childList: true } );
+		};
+		const stopObserver = () => {
+			observer.disconnect();
+		};
+		const overrideSaveButtonHover = ( selector: string ) => {
+			const button = document.querySelector( selector );
+			if ( button ) {
+				button.addEventListener( 'mouseover', startObserver );
+				button.addEventListener( 'mouseout', stopObserver );
+			}
+		};
+
+		if ( canvasMode === 'view' ) {
+			overrideSaveButtonClick( SAVE_HUB_SAVE_BUTTON_SELECTOR );
+			overrideSaveButtonHover( SAVE_HUB_SAVE_BUTTON_SELECTOR );
+			return;
+		}
+		if ( canvasMode === 'edit' ) {
+			overrideSaveButtonClick( HEADER_SAVE_BUTTON_SELECTOR );
+			overrideSaveButtonHover( HEADER_SAVE_BUTTON_SELECTOR );
+			return;
+		}
+
+		return () => {
+			document
+				.querySelector( SAVE_HUB_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'click', saveButtonClickHandler );
+			document
+				.querySelector( HEADER_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'click', saveButtonClickHandler );
+			document
+				.querySelector( SAVE_HUB_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'mouseover', startObserver );
+			document
+				.querySelector( HEADER_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'mouseover', startObserver );
+			document
+				.querySelector( SAVE_HUB_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'mouseout', stopObserver );
+			document
+				.querySelector( HEADER_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'mouseout', stopObserver );
+		};
+	}, [ canvasMode, setIsThemeUpgradeModalOpen ] );
+
+	useEffect( () => {
+		// This overrides the keyboard shortcut (⌘S) for saving.
+		const overrideSaveButtonKeyboardShortcut = ( e: KeyboardEvent ) => {
+			if ( e.key === 's' && ( e.metaKey || e.ctrlKey ) ) {
+				e.preventDefault();
+				e.stopPropagation();
+				setIsThemeUpgradeModalOpen( true );
+			}
+		};
+		document.addEventListener( 'keydown', ( e: KeyboardEvent ) => {
```

Comment:
Should this be just:
```suggestion
		document.addEventListener( 'keydown',  overrideSaveButtonKeyboardShortcut );
```

Reply:
🤦